packages/dbml-core/src/connectors/mysqlConnector.js

```diff
@@ -0,0 +1,509 @@
+import { createConnection } from 'mysql2/promise';
+import { flatten } from 'lodash';
+
+const NUMBER_REGEX = '^-?[0-9]+(.[0-9]+)?$';
+
+/**
+ * @param {string} connection
+ * @returns {Promise<import('mysql2').Connection>} client
+ */
+async function connectMySQL (connection) {
+  let client = null;
+  try {
+    client = await createConnection(connection);
+
+    await client.connect();
+    await client.query('SELECT 1');
+
+    return client;
+  } catch (error) {
+    console.error('MySQL connection error:', error);
+    await client?.end();
+    throw error;
+  }
+}
+
+function getEnumName (tableName, columnName) {
+  return `${tableName}_${columnName}_enum`;
+}
+
+function getGenerationExpression (extraType, generationExpression) {
+  if (extraType === 'VIRTUAL GENERATED') {
+    return `GENERATED ALWAYS AS (${generationExpression}) VIRTUAL`;
+  }
+
+  if (extraType === 'STORED GENERATED') {
+    return `GENERATED ALWAYS AS (${generationExpression}) STORED`;
+  }
+
+  // for timestamp data type
+  if (extraType.includes('on update CURRENT_TIMESTAMP')) {
```

Comment:
I think checking the CURRENT_TIMESTAMP is not enough. For example: NOW(), INTERVAL, ...

Reply:
Ye, I'll add more test cases for this a.
FYI, the checking value is defined by mysql like this

![SmartSelect_20240809-174105_Chrome.jpg](https://github.com/user-attachments/assets/b8d24cfd-ed84-4ad2-b4f4-127966eb6eb1)

Reply:
I added more test in this commit https://github.com/holistics/dbml/pull/602/commits/f0dc9d52059958f8e1042c98657b695762d8ef2c

Also, the `on update` clause can be used for datetime or timestamp; possible values are `CURRENT_TIMESTAMP` and its alias (`now()`, `localtime()`, etc.)

Reply:
So, you should also check the auto-increment attribute?

Reply:
Actually, I checked it in this section 
![image](https://github.com/user-attachments/assets/c0d41375-2399-48ee-b979-99347152b0e6)
Since it is usefull for the increment attribute of DBML field.